upi/azure/02_storage.json

```diff
@@ -1,41 +1,159 @@
 {
-  "$schema" : "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion" : "1.0.0.0",
-  "parameters" : {
-    "baseName" : {
-      "type" : "string",
-      "minLength" : 1,
-      "metadata" : {
-        "description" : "Base name to be used in resource names (usually the cluster's Infra ID)"
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "architecture": {
+      "type": "string",
+      "metadata": {
+        "description": "The architecture of the Virtual Machines"
+      },
+      "defaultValue": "x64",
+      "allowedValues": [
+        "Arm64",
+        "x64"
+      ]
+    },
+    "baseName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Base name to be used in resource names (usually the cluster's Infra ID)"
+      }
+    },
+    "imageRelease": {
+      "type": "string",
+      "metadata": {
+        "description": "RHCOS release image version - used when creating the image definition in the gallery"
       }
     },
-    "vhdBlobURL" : {
-      "type" : "string",
-      "metadata" : {
-        "description" : "URL pointing to the blob where the VHD to be used to create master and worker machines is located"
+    "storageAccountId": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the Storage Account"
+      }
+    },
+    "vhdBlobURL": {
+      "type": "string",
+      "metadata": {
+        "description": "URL pointing to the blob where the VHD to be used to create master and worker machines is located"
       }
     }
   },
-  "variables" : {
-    "location" : "[resourceGroup().location]",
-    "imageName" : "[concat(parameters('baseName'), '-image')]"
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "galleryName": "[concat('gallery_', replace(parameters('baseName'), '-', '_'))]",
+    "imageName": "[parameters('baseName')]",
+    "imageNameGen2": "[concat(parameters('baseName'), '-gen2')]",
+	"imageVersion": "[take(parameters('imageRelease'), len(parameters('imageRelease'))-6)]"
```

Comment:
i'm now thinking if this should not be done here but we should give control to the user. like IPI, we are not making the decision here, but it depends on who is using the template. sure, if they use the release version, this is valid, but a user may choose differently and then it might break.

Reply:
Users are free to edit the templates, but if you think there is more value in exposing this choice, I can change it. 

It had to be like this at first because of MAO expectations from https://github.com/openshift/installer/blob/master/pkg/asset/machines/azure/machines.go#L120 but with https://github.com/openshift/installer/pull/6694 we're now moving to using `latest` in the VM, so the exact value of this variable doesn't matter anymore.

Reply:
I changed it to use the `imageRelease` directly and dropped the `imageVersion` variable.